File: antd-py/examples/09_vaults.py
"""Example 09: Vault store and retrieve.

Vaults provide private encrypted storage keyed by a secret key.
Data is encrypted client-side before being stored on the network.
"""

import os

from antd import AntdClient

client = AntdClient()

# Generate a random secret key for the vault
secret_key = os.urandom(32).hex()

# Store data in the vault
payload = b"Secret vault data that is encrypted"
content_type = 42  # application-defined type
cost = client.vault_put(secret_key, payload, content_type)
print(f"Vault store cost: {cost} atto tokens")

# Retrieve from vault
vault = client.vault_get(secret_key)
print(f"Content type: {vault.content_type}")
print(f"Data: {vault.data.decode()}")

assert vault.data == payload, "Vault round-trip mismatch!"
assert vault.content_type == content_type, "Content type mismatch!"

print("Vault round-trip OK!")

File: antd-js/examples/09-vaults.ts
/**
 * Example 09: Vault store and retrieve.
 *
 * Vaults provide private encrypted storage keyed by a secret key.
 * Data is encrypted client-side before being stored on the network.
 */

import { randomBytes } from "node:crypto";
import { createClient } from "../src/index.js";

const client = createClient();

// Generate a random secret key for the vault
const secretKey = randomBytes(32).toString("hex");

// Store data in the vault
const payload = Buffer.from("Secret vault data that is encrypted");
const contentType = 42; // application-defined type
const cost = await client.vaultPut(secretKey, payload, contentType);
console.log(`Vault store cost: ${cost} atto tokens`);

// Retrieve from vault
const vault = await client.vaultGet(secretKey);
console.log(`Content type: ${vault.contentType}`);
console.log(`Data: ${vault.data.toString()}`);

if (!vault.data.equals(payload)) throw new Error("Vault round-trip mismatch!");
if (vault.contentType !== contentType) throw new Error("Content type mismatch!");

console.log("Vault round-trip OK!");
